sdk/core/azure-core-test/README.md

```diff
@@ -1,108 +1,289 @@
+    - [Run tests with out-of-repo recordings](#run-tests-with-out-of-repo-recordings)
+    - [Sanitize secrets](#sanitize-secrets)
+- [Troubleshooting](#troubleshooting)
+- [Next steps](#next-steps)
+- [Contributing](#contributing)
 
 ## Getting started
 
 To use this package, add the following to your _pom.xml_.
 
 [//]: # ({x-version-update-start;com.azure:azure-core-test;current})
+
 ```xml
+
 <dependency>
-  <groupId>com.azure</groupId>
-  <artifactId>azure-core-test</artifactId>
-  <version>1.18.0</version>
+    <groupId>com.azure</groupId>
+    <artifactId>azure-core-test</artifactId>
+    <version>1.18.0</version>
 </dependency>
 ```
+
 [//]: # ({x-version-update-end})
 
 ## Key concepts
 
-* [TestBase][TestBase.java]: Base test class that creates an `InterceptorManager` for each unit test and either plays
-  back test session data or records the test session. Each session's file name is determined using the name of the test.
-* [InterceptorManager][InterceptorManager.java]: Main class used to record and playback `NetworkCallRecords`.
+* Run tests in `Record` mode:
+  To record means to intercept any HTTP request, store it in a file, then store the response received from the live
+  resource that was originally targeted.
+* Run tests in `Playback` mode:
+  To playback means to intercept any HTTP request and to respond it with the stored response of a previously recorded
+  matching request.
+* Run tests in `Live` mode:
+  To run live means to not intercept any HTTP request and to send them to the live resource.
+* Sanitize sensitive information:
+  Sensitive information means content like passwords, unique identifiers or personal information should be cleaned up
+  from the recordings.
+* [TestProxyTestBase][TestProxyTestBase.java]: Base test class that creates an `InterceptorManager` and enables the test
+  to use [`test-proxy`][test-proxy-readme] for running the test. It either plays back test session data or records the
+  test session.
+* [InterceptorManager][InterceptorManager.java]: A class that keeps track of network calls by either reading the data
+  from an existing test session record or recording the network calls in memory. Test session records are saved or read
+  from "<i>.assets/{library-level}/src/test/resources/session-records/TestFileName.testName}.json</i>".
+* [TestProxyRecordPolicy][TestProxyRecordPolicy.java]: Pipeline policy that records network calls using
+  the [`test-proxy`][test-proxy-readme].
+* [TestProxyPlaybackClient][TestProxyPlaybackClient.java]: HTTP client that plays back responses from the recorded data
+  of session-record using test proxy.
+
+## Write or run tests
+
+Newer SDK tests utilize the [Azure SDK Tools Test Proxy][test-proxy-readme] to record and playback HTTP interactions.
+To migrate from existing [TestBase][TestBase.java] to use the test proxy, or to learn more about using the test proxy,
+refer to the [test proxy migration guide][test-proxy-migration-guide].
```

Comment:
This is already in the introduction section and we don't need to repeat this here.